api/app/controllers/spree/api/v2/platform/products_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Spree
+  module Api
+    module V2
+      module Platform
+        class ProductsController < ::Spree::Api::V2::ResourceController
+          private
+
+          def sorted_collection
+            collection_sorter.new(collection, current_currency, params, allowed_sort_attributes).call
+          end
+
+          def collection
+            @collection ||= collection_finder.new(scope: scope, params: params, current_currency: current_currency).execute
```

Comment:
@MatthewKennedy let's not use custom finders in Platform API at all and rely on ransack